src/LsmTree.hs

```diff
@@ -26,14 +25,15 @@ emptyTree = Leaf
 initTree :: Ord a => a -> Tree a
 initTree x = TreeNode Leaf x Leaf
 
-insertTree :: Ord a => Tree a -> a -> Tree a
-insertTree Leaf x = initTree x
-insertTree (TreeNode l val r) x
-   | x < val   = TreeNode (insertTree l x) val r
-   | otherwise = TreeNode l val (insertTree r x)
+insertTree :: Ord a => a -> Tree a -> Tree a
+insertTree x tree = case tree of
+    Leaf             -> initTree x
+    TreeNode l val r -> case x < val of
```

Comment:
Instead of pattern-matching on `x < val` result, it's better to pattern match on `case compare x val` for more idiomatic code